todo-app/src/stores/TodoStore.js

```diff
@@ -0,0 +1,54 @@
+import { action, observable } from 'mobx';
+import moment from 'moment';
+
+// Data
+import TodoData from '../data/todo.json';
+
+export class TodoStore {
+  @observable todos = TodoData
+  @observable newDescription = ''
+  @observable newDeadline = moment().format('YYYY-MM-DD')
+
+  // Change the state of DONE foreach todo
+  @action
+  handelCheck = (todo) => {
+    todo.done = !todo.done
+  }
+
+  @action
+  handelDeleteItem = (todo) => {
+    this.todos = this.todos.filter((item) => {
+      return item.id !== todo.id
+    })
+  }
+
+  @action
+  handleDescriptionChange = (event) => {
+    this.newDescription = event.target.value
+  }
+
+  @action
+  handleDateChange = (event) => {
+    this.newDeadline = event.target.value
+  }
+
+  // add new todo
+  @action
+  handleSubmit = (event) => {
+    event.preventDefault();
```

Comment:
In my opinion, using `event.preventDefault` in a MobX action is incorrect. This is of course somewhat an opinion, but MobX actions should be totally independent of the UI. They should be (ideally) pure functions that take an input and mutate state based on that input.

If you want to use a form and call `preventDefault`, it is probably better to do this outside your MobX action, and just pass the value in here.

It actually makes me realise we probably shouldn't be passing `event` objects into these MobX actions in the first place, and should only pass the actual values. I will try to remember to discuss this in class.